ts/Data/DataTableCore.ts

```diff
@@ -96,19 +106,31 @@ class DataTableCore {
          * @name Highcharts.DataTable#id
          * @type {string}
          */
-        this.id = (options.id || uniqueKey());
+        this.id = (id || uniqueKey());
         this.modified = this;
         this.rowCount = 0;
         this.versionTag = uniqueKey();
-        this.columnNames = options.columnNames;
-        this.firstRowAsNames = options.firstRowAsNames;
-        this.orientation = options.orientation;
-        this.dataModifier = options.dataModifier;
-        this.beforeParse = options.beforeParse;
+
+        // Set the table options only if defined.
+        if (defined(columnNames)) {
+            this.columnNames = columnNames;
+        }
+        if (defined(firstRowAsNames)) {
+            this.firstRowAsNames = firstRowAsNames;
+        }
+        if (defined(orientation)) {
+            this.orientation = orientation;
+        }
+        if (defined(dataModifier)) {
+            this.dataModifier = dataModifier;
+        }
+        if (defined(beforeParse)) {
+            this.beforeParse = beforeParse;
+        }
```

Comment:
First of all, it seems that these fields should not be there at all on `DataTableCore`. If we want to store them here, they should be in `DataTable`.